### DISCUSSION TITLE
HomeAssistant integration

### DISCUSSION THREAD

**Discussion by karbobrak** [jonassjoh/dspW245#15] [Q&A]
Hi, newbie here. 
I have a DSP-W115-2211
Could you please help me out a little bit? I took your script and copied it into config/python_scripts . 
When I try to run the script from services, I get the following error
`Error loading script dspw245.py: 
Line 213: "_recv" is an invalid variable name because it starts with "_", 
Line 216: "_recv" is an invalid variable name because it starts with "_", 
Line 224: Augmented assignment of object items and slices is not allowed., 
Line 360: "__name__" is an invalid variable name because it starts with "_"`
Furthermore, even if I'd managed to run it, what is the next step? How to turn the plug on / off from homeassistant?
EDIT: as I found out, I can run python script as actions in automatizations, so does this mean, that once I manage to run the script, the plug will turn on? Do I need to scripts, with two different lines towards the end of the script a) sp.set_socket(1, on=True) and b) sp.set_socket(1, on=False)

**Comment by jonassjoh** [comment]
Hi!

I'm unfortunally not familiar with homeassistant and I cannot help you with anything related to that.

I did however, run into this thread some time ago (https://community.home-assistant.io/t/dlink-smart-plug-wi-fi-dsp-w115/53760/23) perhaps it will give you some guidance. There appears to be a guide of sorts later in the thread, perhaps that's what you're looking for.

Again, I'm not familiar with HA, but you could also try the Node.js version of this by Garfonso https://github.com/Garfonso/dlinkWebSocketClient 

I've also never before seen that error, or anything similar to it previously. It got me a little curious and it looks like it's a restriction of RestrictedPython, which I'm going to go ahead and assume is being used somewhere. Noone seemed to mention an issue regarding this in the thread I linked, but if you were to keep running into issues because of those names, you could try eg. `_recv` to `recv`, `data[i] += 255` to `data[i] = data[i] + 255` (?) and you could remove everything from like 360 and down since it's only meant to act as an example anyway.

If you still have any questions feel free to ask, I'm not sure if I'll be of any help, but who knows :p

Good luck.
